src/Event/Decorator/AbstractDecorator.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * CakePHP : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP Project
+ * @since         3.3.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Event\Decorator;
+
+abstract class AbstractDecorator
+{
+
+    /**
+     * Callable
+     *
+     * @var callable
+     */
+    protected $_callable;
+
+    /**
+     * Decorator options
+     *
+     * @var array
+     */
+    protected $_options = [];
+
+    /**
+     * Constructor.
+     *
+     * @param callable $callable Callable.
+     * @param array $options Decorator options.
+     */
+    public function __construct(callable $callable, array $options = [])
+    {
+        $this->_callable = $callable;
+        $this->_options = $options;
+    }
+
+    /**
+     * Invoke
+     *
+     * @link http://php.net/manual/en/language.oop5.magic.php#object.invoke
+     * @return mixed
+     */
+    public function __invoke()
+    {
+        return $this->_call($this->_callable, func_get_args());
```

Comment:
@burzum You shouldn't pass the first param since the `_call` just expect the args now.